doc/release/1.12.0-notes.rst

```diff
@@ -23,6 +23,20 @@ Future Changes
 Compatibility notes
 ===================
 
+DeprecationWarning to error
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Indexing with floats raises IndexError,
+  e.g., a[0, 0.0].
+* Indexing with non-integer array_like raises ``IndexError``,
+  e.g., ``a['1', '2']``
+* Indexing with multiple ellipsis raises ``IndexError``,
+  e.g., ``a[..., ...]``.
+* Indexing with boolean where integer expected raises ``IndexError``,
+  e.g., ``a[False:True:True]``.
```

Comment:
@charris changing would mean putting Nathaniels cleanup already in now (but without getting rid of the error replacement). Frankly, it does not bother me much to give a warning that won't be acted up on, it is a rather weird thing to use bools there, doesn't mean it can't annoy someone else ;).